benchmarks/distributed_contact/distributed_contact.cpp

```diff
@@ -0,0 +1,235 @@
+{
+  using Triangle = ArborX::Triangle<DIM, Coordinate>;
+
+  int comm_rank;
+  MPI_Comm_rank(comm, &comm_rank);
+  int comm_size;
+  MPI_Comm_size(comm, &comm_size);
+
+  int const num_triangles = triangles.size();
+
+  Coordinate const d = 0.8;
+  int rows_per_dim = 40;
+
+  int N = std::cbrt(comm_size);
+  int Nx = comm_rank / (N * N);
+  int Ny = (comm_rank - Nx * (N * N)) / N;
+  int Nz = comm_rank - Nx * (N * N) - Ny * N;
+
+  auto rank_offset_x = Nx * rows_per_dim * d;
+  auto rank_offset_y = Ny * rows_per_dim * d;
+  auto rank_offset_z = Nz * rows_per_dim * d;
+
+  std::vector<Triangle> primitives;
+  std::vector<Triangle> predicates;
+
+  auto shift_point = [](std::array<Coordinate, DIM> const &offset,
+                        ArborX::Point<DIM, Coordinate> const &point) {
+    return ArborX::Point<DIM, Coordinate>{
+        point[0] + offset[0], point[1] + offset[1], point[2] + offset[2]};
+  };
+
+  primitives.reserve(3 * rows_per_dim * rows_per_dim);
+  predicates.reserve(3 * rows_per_dim * rows_per_dim);
+  for (int i = 0; i < rows_per_dim; ++i)
+    for (int j = 0; j < rows_per_dim; ++j)
+      for (int k = 0; k < rows_per_dim; ++k)
+      {
+        if (i > 0 && i < rows_per_dim - 1 && j > 0 && j < rows_per_dim - 1 &&
+            k > 0 && k < rows_per_dim - 1)
+          continue; // Skip interior points
+
+        std::array<Coordinate, DIM> offset = {rank_offset_x + i * d,
+                                              rank_offset_y + j * d,
+                                              rank_offset_z + k * d};
+
+        if ((i + j + k) % 2 == 0)
+          for (int t = 0; t < num_triangles; ++t)
+            predicates.emplace_back(
+                Triangle{shift_point(offset, points[triangles[t][0]]),
+                         shift_point(offset, points[triangles[t][1]]),
+                         shift_point(offset, points[triangles[t][2]])});
+        else
+          for (int t = 0; t < num_triangles; ++t)
+            primitives.emplace_back(
+                Triangle{shift_point(offset, points[triangles[t][0]]),
+                         shift_point(offset, points[triangles[t][1]]),
+                         shift_point(offset, points[triangles[t][2]])});
+      }
+
+  return std::make_pair(primitives, predicates);
+}
+
+// Callback to store the result indices
+struct ExtractIndex
+{
+  template <typename Query, typename Value, typename Output>
+  KOKKOS_FUNCTION void operator()(Query const &, Value const &value,
+                                  Output const &out) const
+  {
+    out(value.index);
+  }
+};
+
+template <typename Coordinate>
+int main_(MPI_Comm const comm)
+{
+  ArborXBenchmark::TimeMonitor time_monitor;
+
+  using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+  using MemorySpace = typename ExecutionSpace::memory_space;
+
+  using Box = ArborX::Box<DIM, Coordinate>;
+
+  auto [points, triangles] = readTriangles<Coordinate>("unit_sphere.txt");
```

Comment:
How did you create `unit_sphere.txt`? It would be nice to add a `README` or a comment to  know how to recreate this file if necessary

Reply:
> How did you create `unit_sphere.txt`? It would be nice to add a `README` or a comment to know how to recreate this file if necessary

I didn't. I grabbed it from the repository used in that paper.